ksqldb-engine/src/main/java/io/confluent/ksql/function/UdafTypes.java

```diff
@@ -54,38 +56,43 @@
   private final Type inputType;
   private final Type aggregateType;
   private final Type outputType;
-  private final String functionInfo;
+  private final List<ParameterInfo> literalParams;
   private final String invalidClassErrorMsg;
   private final SqlTypeParser sqlTypeParser;
 
   UdafTypes(
       final Method method,
-      final String functionInfo,
+      final FunctionName functionName,
       final SqlTypeParser sqlTypeParser
   ) {
-    Objects.requireNonNull(method);
-    this.functionInfo = Objects.requireNonNull(functionInfo);
     this.invalidClassErrorMsg = "class='%s'"
         + " is not supported by UDAFs. Valid types are: " + SUPPORTED_TYPES + " "
-        + functionInfo;
+        + Objects.requireNonNull(functionName, "functionName");
     final AnnotatedParameterizedType annotatedReturnType
         = (AnnotatedParameterizedType) method.getAnnotatedReturnType();
     final ParameterizedType type = (ParameterizedType) annotatedReturnType.getType();
     this.sqlTypeParser = Objects.requireNonNull(sqlTypeParser);
 
-    inputType = type.getActualTypeArguments()[0];
-    aggregateType = type.getActualTypeArguments()[1];
-    outputType = type.getActualTypeArguments()[2];
+    this.inputType = type.getActualTypeArguments()[0];
+    this.aggregateType = type.getActualTypeArguments()[1];
+    this.outputType = type.getActualTypeArguments()[2];
```

Comment:
nit: not your code, but since you're already changing this can we call `type.getActualTypeArguments()` just once (and also check that the size is exactly three)